tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java

```diff
@@ -119,34 +156,25 @@ public void resourceChanged( final IResourceChangeEvent event )
         try
         {
             final IResourceDelta delta = event.getDelta();
-            final SDK sdk = SDKUtil.getWorkspaceSDK();
 
-            if( delta != null && sdk != null )
+            if( delta != null )
             {
                 for( IResourceDelta child : delta.getAffectedChildren() )
                 {
-                    final IPath deltaLocation = child.getResource().getLocation();
+                    final IResourceDelta[] sdkChangedFiles =
```

Comment:
This code matches too much.  What if we have a project that is in not way related to the SDK or SDK workspace project and it has a file called "build.properties" in it.  this pattern matcher is going ot match isn't it?

Please be careful to think of other situations that don't exactly match the specific thing you are coding for and see if you can break your own code assumptions.  

You just named your variable "sdkChangedFiles"  but you don't know that... the files could be any resource in the workspace that has changed... like "myFooProject" and it in it has a "myFooProject/build.properties" file.  That isn't an "sdkChangedFile".